src/api/EventsAPI.ts

```diff
@@ -19,30 +16,101 @@ export type EventObject = {
   staffPointBonus: number;
+  facebookUrl?: string;
 };
 
-const handleErrors = (response: Response): Promise<EventsResponse> => {
+export type EventsArray = EventObject[];
+
+export type EventsResponse = {
+  error: unknown;
+  events: EventsArray;
+};
+
+export type EventResponse = {
+  error: unknown;
+  event: EventObject;
+};
+
+const handleErrors = (response: Response) => {
   if (!response.ok) {
     throw Error(response.statusText);
   }
   return response.json();
 };
 
-const getAllEvents = async (): Promise<EventsArray | undefined> => {
-  let apiurl = "https://api.acmucsd.com/api/v2/event/future";
-
-  // TODO: Fix test data for development purposes (images are not showing up)
-  // if (!process.env.NODE_ENV || process.env.NODE_ENV === "development") {
-  //   apiurl = "https://testing.api.acmucsd.com/api/v2/event/future"
-  // }
+const getAllEvents = async (
+  type: "past" | "future" | "" = ""
+): Promise<EventsArray | undefined> => {
+  const api_url = `${EVENT_API}/${type}`;
 
   try {
-    const response: Response = await fetch(apiurl);
+    const response: Response = await fetch(api_url);
     const result: EventsResponse = await handleErrors(response);
     return result.events;
   } catch (error) {
     return undefined;
   }
 };
 
-export { getAllEvents };
+const getEvent = async (uuid: string): Promise<EventObject | undefined> => {
+  let api_url = `${EVENT_API}/${uuid}`;
+
+  try {
+    const response: any = await fetch(api_url);
+    const result: EventResponse = await handleErrors(response);
+    return result.event;
+  } catch (error) {
+    return undefined;
+  }
+};
+
+const createDownloadFile = (textContent: string, title: string): void => {
+  const element = document.createElement("a");
+  const file = new Blob([textContent], { type: "text/plain" });
+  element.href = URL.createObjectURL(file);
+  element.download = `${title}.ics`;
+  document.body.appendChild(element);
+  element.click();
+};
+
+const downloadICS = (event: EventObject): void => {
+  const startDate = new Date(event.start);
+  const endDate = new Date(event.end);
+  const duration = endDate.getTime() - startDate.getTime();
+
+  const attributes: ics.EventAttributes = {
+    start: [
+      startDate.getFullYear(),
+      startDate.getMonth() + 1,
+      startDate.getDate(),
+      startDate.getHours(),
+      startDate.getMinutes(),
+    ],
+    duration: { minutes: duration / (1000 * 60) },
+    title: event.title,
+    description: event.description,
+    location: event.location,
+    url: `https://acmucsd.com/events/${formatURLEventTitle(event.title)}-${event.uuid}`,
+    organizer: { name: "ACM at UCSD", email: "acm@ucsd.edu" },
+  };
+
+  const response = ics.createEvent(attributes);
+
+  createDownloadFile(response.value, formatURLEventTitle(event.title));
+};
+
+const saveToGoogleCal = ({ title, description, location, start, end }: EventObject): void => {
+  const url = new URL("https://www.google.com/calendar/render?action=TEMPLATE");
```

Comment:
link to the template we're using for this; there's always a chance google deprecates this so it'll be good to know where we got this pattern from (if you can find an api reference that's even better)